### PR TITLE
chore(Storybook): migrate to version 6

### PR DESCRIPTION
**Pull request by eszthoff**
Pending issues after the upgrade:
* CSS loaded in the wrong order (e.g. see Pagination story)
* Console logs no longer redirected to Actions tab. [Addon doesn't support v6](https://github.com/storybookjs/storybook-addon-console/issues/53). No solution at this time
* migrating text only stories (ArrangingOfZIndices) is incomplete. Tried to move to `.mdx` file, but was not loaded to the UI.
* should we update AutosuggestDeprecated story?

### CODE REVIEW

**Comment by antipin** [comment]
Previous attempt: https://github.com/textkernel/oneui/pull/679

File: .storybook/main.js [hunk 1]
```diff
@@ -0,0 +1,24 @@
+const path = require('path');
+const { getRules } = require('../scripts/build/webpack.config');
+
+module.exports = {
+    stories: ['../stories/**/*.@(js|jsx|ts|tsx)'],
```
[the remark this answers is not in the document]
Comment (eszthoff) [reply .storybook/main.js:5]: yes, I tried that. But that didn't help with loading them. I think it is better to have that sorted out in the other ticket for it.

File: src/components/DatePicker/DatePicker.tsx [hunk 2]
```diff
@@ -6,6 +6,14 @@ import './DatePicker.scss';
...
 const DEFAULT_YEAR_RANGE = 100;
 
+/**
+ * ## Usage information
+ * This component is a thin wrapper around [react-datepicker](https://github.com/Hacker0x01/react-datepicker/),
+ * only setting the header element and add css.
+ *
+ * You can pass [props](https://github.com/Hacker0x01/react-datepicker/blob/master/docs/datepicker.md) according to their definition.
+ * See also [examples](https://reactdatepicker.com/) on their website.
+ */
```
[the remark this answers is not in the document]
Comment (eszthoff) [reply src/components/DatePicker/DatePicker.tsx:16]: yes, me too. :)

File: src/components/List/List.tsx [hunk 3]
```diff
@@ -3,7 +3,7 @@ import { bem, scrollIntoViewIfNeeded } from '../../utils';
 import { LIST_NAVIGATION_DIRECTIONS, ENTER_KEY } from '../../constants';
 import styles from './List.scss';
 
-interface Props extends React.HTMLAttributes<HTMLUListElement> {
+export interface Props extends React.HTMLAttributes<HTMLUListElement> {
```
[the remark this answers is not in the document]
Comment (eszthoff) [reply src/components/List/List.tsx:6]: I can go through and rename them, just let me know if you want me to. Or is it better to come up with a standard in the other ticket maybe.

File: .storybook/preview-head.html [hunk 4]
```diff
@@ -1,8 +0,0 @@
-<style>
-    body {
```
Comment (eszthoff) [inline .storybook/preview-head.html:2]: I forgot to migrate this bit. Maybe that's why the font is the wrong one.

File: .eslintrc.json [hunk 5]
```diff
@@ -84,7 +84,9 @@
                 "react/display-name": "off",
```
Comment (mukiienko) [inline .eslintrc.json:84]: I think in the current implementation we don't need `stories/index.ts`

File: .storybook/main.js (hunk 1 above)
Comment (mukiienko) [inline .storybook/main.js:5]: In the old version loading files was implemented via `index.ts` and we had a right overriding because stories were ordered in the right way in the `index.ts`. In the current implementation we still can control order by items in the array `stories`, but first we need to group our stories into subgroups/subfolders `Concepts`, `Theme`, `Atoms` etc (in the same way as we have for preview.js), for example:
```
 stories: ['../stories/theme/*.@(js|jsx|ts|tsx)', '../stories/**/*.@(js|jsx|ts|tsx)'],
```
It fixed the issue in Pagination story:
Screenshot 2022-08-19 at 13 19 22

File: src/components/List/List.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (mukiienko) [reply src/components/List/List.tsx:6]: Let's do that in an other ticket.

File: .storybook/main.js (hunk 1 above)
Comment (eszthoff) [reply .storybook/main.js:5]: I'm sure I tried this before, and it didn't work. But you are right, and it does work. Nice one!!!
I'll push a commit with this.

File: .eslintrc.json (hunk 5 above)
Comment (eszthoff) [reply .eslintrc.json:84]: removed it now, that the folders import works

File: .storybook/main.js (hunk 1 above)
Comment (eszthoff) [reply .storybook/main.js]: done

File: stories/organisms/AutosuggestDeprecated.js [hunk 6]
```diff
@@ -0,0 +1,184 @@
+// import * as React from 'react';
```
Comment (andrentk) [inline stories/organisms/AutosuggestDeprecated.js:1]: Is this file still needed?

File: .storybook/main.js (hunk 1 above)
Comment (mukiienko) [reply .storybook/main.js]: We still have a problem with overriding styles for `Pagination`, because these styles come from `Themeroller` that comes after `Pagination`, we need to put it on before high-level components:
```
    stories: [
        '../stories/theme/*.@(js|jsx|ts|tsx)',
        '../stories/atoms/*.@(js|jsx|ts|tsx)',
        '../stories/molecules/*.@(js|jsx|ts|tsx)',
        '../stories/organisms/*.@(js|jsx|ts|tsx)',
        '../stories/packages/*.@(js|jsx|ts|tsx)',
        '../stories/**/*.@(js|jsx|ts|tsx)',
    ],
```

File: stories/organisms/AutosuggestDeprecated.js (hunk 6 above)
Comment (eszthoff) [reply stories/organisms/AutosuggestDeprecated.js:1]: I was wondering about that. I was thinking a bit to keep it for internal reference. But perhaps that is not needed

File: .storybook/main.js (hunk 1 above)
Comment (eszthoff) [reply .storybook/main.js]: I'm actually not convinced that the theme should be at the beginning. Its whole purpose is to overwrite the styles, and then storybook will showcase the components as they will look like if adding that theme to the application. What do you think?

File: stories/organisms/AutosuggestDeprecated.js (hunk 6 above)
Comment (eszthoff) [reply stories/organisms/AutosuggestDeprecated.js]: removed it

File: .storybook/main.js (hunk 1 above)
Comment (eszthoff) [reply .storybook/main.js]: done

**Review by andrentk — APPROVED** [review]
